scripts/build.py

```diff
@@ -1177,6 +1177,40 @@ def compile_typescript_files(project_dir):
     subprocess.check_call(cmd, shell=True)
 
 
+def compile_typescript_files_in_watch_mode(project_dir):
+    """Compiles typescript files in watch mode to produce javascript
+    files in local_compiled_js folder.
+
+    Args:
+        project_dir: str. The project directory which contains the ts files
+            to be compiled.
+    """
+    kill_cmd = (
+        'kill `ps aux | grep "[.]./node_modules/typescript/bin/tsc --project . '
+        '--watch" | awk \'{print $2}\'`'
+    )
+    subprocess.call(kill_cmd, shell=True, stdout=subprocess.PIPE)
+    print 'Compiling ts files in watch mode...'
+    cmd = '../node_modules/typescript/bin/tsc --project %s --watch' % (
+        project_dir)
+    subprocess.Popen('%s > tsc_output_log.txt' % cmd, shell=True)
```

Comment:
I don't think you need shell=True here? Try to avoid using shell=True where possible, it has security implications.

Reply:
I tried removing it but strangely on splitting the command, typescript gives an error that source files cannot be specified with the project dir option (tsc_output_log.txt is being treated as a source file)

Reply:
That seems weird, I think it's worth looking into that further.

Reply:
Sure, I am looking into it.

Reply:
This is fixed now, PTAL.